packages/openapi3/src/openapi.ts

```diff
@@ -1161,6 +1161,8 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
             return applyIntrinsicDecorators(cadlType, { type: "integer", format: "int32" });
           case "int64":
             return applyIntrinsicDecorators(cadlType, { type: "integer", format: "int64" });
+          case "safeint":
+            return applyIntrinsicDecorators(cadlType, { type: "integer", format: "int64" });
```

Comment:
Hmm. Should we add `{ minimium: Number.MIN_SAFE_INTEGER, maximum: number: Number.MAX_SAFE_INTEGER }` or some other distinction from int64? I suppose if we do the min/max, then we have to be careful to take higher explicit min, lower explicit max from user?

Reply:
I know I'm being pedantic here, but I don't think `int53` would be quite right. I take `int32` et al to be two's complement encoded. So the range for `int32`, for example, is `-2^31` to `2^31-1`. But the safe integer range is `-2^53-1` to `2^53-1` and not `-2^52` to `-2^52-1` . So `safeint` is nearly `int54`, but different from that too since `-2^53` is not allowed.

I'm not sure about leaving the format off either, but I like it better than `intNN` for this special case.

Reply:
On further thought, I want to argue for `int64`.  In JSON Schema, `format` has no formal meaning -- it is up to implementations to define and interpret.

https://datatracker.ietf.org/doc/html/draft-fge-json-schema-validation-00#section-7

For Azure, we use `format` to guide our code generators. And for every language I can think of, the code that we want to produce for a `safeint` is exactly the same code that we produce for `format: int64`.

So while perhaps not technically correct, I think we want `format: int64` because our code generators will understand this and do the right thing.